domains/certificates/Query.API/API/MeasurementsSyncer/MeasurementsSyncService.cs

```diff
@@ -3,67 +3,58 @@
-    {
-        _logger = logger;
-        _slidingWindowState = slidingWindowState;
-        _measurementsClient = measurementsClient;
-        _slidingWindowService = slidingWindowService;
-        _measurementSyncMetrics = measurementSyncMetrics;
-        _measurementSyncPublisher = measurementSyncPublisher;
-        _meteringPointsClient = meteringPointsClient;
-    }
+    private readonly MeasurementsSyncOptions _options = options.Value;
 
     public async Task FetchAndPublishMeasurements(MeteringPointSyncInfo syncInfo, MeteringPointTimeSeriesSlidingWindow slidingWindow,
         CancellationToken stoppingToken)
     {
         var synchronizationPoint = UnixTimestamp.Now().RoundToLatestHour();
         var fetchedMeasurements = await FetchMeasurements(slidingWindow, syncInfo.MeteringPointOwner, synchronizationPoint, stoppingToken);
-        var meteringPoints = await _meteringPointsClient.GetOwnedMeteringPointsAsync(new OwnedMeteringPointsRequest() { Subject = syncInfo.MeteringPointOwner });
+        var meteringPoints = await meteringPointsClient.GetOwnedMeteringPointsAsync(new OwnedMeteringPointsRequest() { Subject = syncInfo.MeteringPointOwner });
         var meteringPoint = meteringPoints.MeteringPoints.First(mp => mp.MeteringPointId == slidingWindow.GSRN);
 
-        _measurementSyncMetrics.AddNumberOfMeasurementsFetched(fetchedMeasurements.Count);
+        measurementSyncMetrics.AddNumberOfMeasurementsFetched(fetchedMeasurements.Count);
 
         if (fetchedMeasurements.Count > 0)
         {
-            var measurementsToPublish = _slidingWindowService.FilterMeasurements(slidingWindow, fetchedMeasurements);
+            var measurementsToPublish = slidingWindowService.FilterMeasurements(slidingWindow, fetchedMeasurements, _options.MinimumAgeBeforeIssuingInHours);
 
-            _logger.LogInformation(
+            logger.LogInformation(
                 "Publishing {numberOfMeasurementsLeft} of total {numberOfMeasurements} measurements fetched for GSRN {GSRN}",
                 measurementsToPublish.Count,
                 fetchedMeasurements.Count,
                 slidingWindow.GSRN);
 
             if (measurementsToPublish.Any())
             {
-                await _measurementSyncPublisher.PublishIntegrationEvents(meteringPoint, syncInfo, measurementsToPublish, stoppingToken);
+                await measurementSyncPublisher.PublishIntegrationEvents(meteringPoint, syncInfo, measurementsToPublish, stoppingToken);
             }
 
-            _slidingWindowService.UpdateSlidingWindow(slidingWindow, fetchedMeasurements, synchronizationPoint);
-            await _slidingWindowState.UpsertSlidingWindow(slidingWindow, stoppingToken);
-            await _slidingWindowState.SaveChangesAsync(stoppingToken);
+            slidingWindowService.UpdateSlidingWindow(slidingWindow, fetchedMeasurements, synchronizationPoint);
```

Comment:
Does it work? Have you tested in a preview env? Because you are setting the sync position to now afterwards.